src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -651,6 +660,27 @@ theorem zero_shiftLeft (n : Nat) : 0#w <<< n = 0#w := by
   cases h₁ : decide (i < m) <;> cases h₂ : decide (n ≤ i) <;> cases h₃ : decide (i < n)
   all_goals { simp_all <;> omega }
 
+theorem shiftLeft_xor_distrib (x y : BitVec w) (n : Nat) :
+    (x ^^^ y) <<< n = (x <<< n) ^^^ (y <<< n) := by
```

Comment:
The naming conventions are vague, but I'm sure I've seen infix-ness of operators being reflected in the naming scheme, which would make this `xor_shiftLeft_distrib`.

Furthermore, I don't think there is a big precedent for `_distrib` lemmas, judging from a search through the docs.
See, e.g, [`Nat.add_mul`](https://leanprover-community.github.io/mathlib4_docs/Init/Data/Nat/Basic.html#Nat.add_mul).

Following that, we could call this `xor_shiftLeft`, which matches the preference for shorter names.

(Note, this is just a suggestion following my interpretation of an incomplete naming convention, feel free to go with another name if you find it more fitting, and can somewhat justify so)

Reply:
We already used this name in https://github.com/leanprover/lean4/pull/4667, and there is precedence in `Nat.left_distrib`, `Bool.and_or_distrib_left`, ...

I would prefer to leave it like this. Is that OK?

Reply:
It seems my search for precedent was too cursory. It's for sure best to be consistent with #4667 :+1:

Reply:
Cool.